### PR TITLE
Add support for IBM Granite 3.x models

### PR DESCRIPTION
**Pull request by frreiss**
## Motivation

This PR adds support for inference on non-MoE IBM Granite 3.x models (see https://huggingface.co/collections/ibm-granite/granite-30-language-models-66fdb59bbb54785c3512114f).

Granite 3.0's architecture is similar to that of Llama 2 and 3, but with additional multiplier parameters for improved numerical stability through Maximal Update Parameterization.

## Modifications

* Added new model type for Granite models
* Backported change to `LogitsProcessor` from vLLM 0.4. This change add an optional logits multiplier parameter. Granite models use this multiplier to implement Maximal Update Parameterization.
* Added chat template for `granite-3-instruct` family
* Added 2b parameter Granite model to regression suite

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [x] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/references/contributor_guide.md).
- [x] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Comment by merrymercy** [comment]
Thanks for the contribution. It is merged!
